src/i2c.rs

```diff
@@ -803,7 +803,18 @@ where
             .write(|w| unsafe { w.bits(u32::from(addr) << 1) });
 
         // Wait until address was sent
-        while self.i2c.sr1.read().addr().bit_is_clear() {}
+        while {
+            let sr1 = self.i2c.sr1.read();
+
+            // If we received a NACK, then this is an error
+            if sr1.af().bit_is_set() {
+                self.i2c.sr1.modify(|_, w| w.af().clear_bit());
+                return Err(Error::NACK);
+            }
+
+            // Wait for the address to be acknowledged.
```

Comment:
```suggestion
            // Wait for the address to be acknowledged
```

Reply:
What's you're preferred commenting grammar? I can update any new comments, but I'm unsure how to format them based on this change request.

Reply:
Well, for me full sentences or sentences in a paragraph end on a period. Sentence-like remarks don't. It's likely we're not consistent about that, it's also likely I don't always notice but when I do I tend to nitpick. ;)